src/security/Session.h

```diff
@@ -18,6 +18,31 @@
 #if USE_OPENSSL
 #if HAVE_OPENSSL_SSL_H
 #include <openssl/ssl.h>
+
+/// These are not included by openssl/ssl.h when deprecated API is disabled
+#include <openssl/opensslv.h>
+#include <openssl/bn.h>
+#include <openssl/rsa.h>
+#include <openssl/dh.h>
+#include <openssl/x509.h>
+
+/// Use new calls for API deprecated in OpenSSL 1.1.0
+/// Note that all of the calls in this next group were implemented at the same time
+/// as X509_STORE_CTX_set0_untrusted, in both OpenSSL 1.1.0, and LibreSSL 2.7.0.
+#if !HAVE_LIBCRYPTO_X509_STORE_CTX_SET0_UNTRUSTED
+#define OPENSSL_VERSION SSLEAY_VERSION
+#define OpenSSL_version SSLeay_version
+#define X509_getm_notBefore X509_get_notBefore
+#define X509_getm_notAfter X509_get_notAfter
+#define X509_set1_notBefore X509_set_notBefore
+#define X509_set1_notAfter X509_set_notAfter
+#endif
+/// lhash.h functions, on the other hand, need a check of their own.
```

Comment:
Please document this #if code block independent of anything else going on. As if it was the only thing being done. It should also go in security/forward.h. Entries in that file should be alphabetical order as much as possible, which would make this bit not only separately located to the earlier block, but also before it.

Reply:
This block is only needed in `src/security/cert_generators/file/certificate_db.cc`, so I moved it to `src/security/cert_generators/file/certificate_db.h`.